src/otx/core/data/transform_libs/torchvision.py

```diff
@@ -503,41 +517,74 @@ def __init__(
     ) -> None:
         super().__init__()
 
-        assert scale is not None or scale_factor is not None, "`scale` and`scale_factor` can not both be `None`"  # noqa: S101
+        if resize_size is None or scale_factor is None:
+            msg = "resize_size will be set to the default value because both resize_size and scale_factor are None."
+            logger.warning(msg)
 
-        if scale is None:
-            self.scale = None
-        elif isinstance(scale, int):
-            self.scale = (scale, scale)
+        if resize_size is not None and not (
+            isinstance(resize_size, int)
+            or (isinstance(resize_size, Sequence) and all(isinstance(r, int) for r in resize_size))
+        ):
+            msg = f"resize_size must be int or Sequence[int], but got `{type(resize_size)}`."
+            raise TypeError(msg)
+
+        if scale_factor is None:
+            self.scale_factor = None
+        elif isinstance(scale_factor, float):
+            self.scale_factor = (scale_factor, scale_factor)
+        elif isinstance(scale_factor, Sequence) and not all(isinstance(sf, float) for sf in scale_factor):
+            msg = (
+                "scale_factor must be float or Sequence[float], "
+                f"but got `{type(scale_factor)}[{type(scale_factor[0])}]`."
+            )
+            raise TypeError(msg)
+        elif isinstance(scale_factor, Sequence) and len(scale_factor) == 2:
+            self.scale_factor = tuple(scale_factor)
         else:
-            self.scale = tuple(scale)  # type: ignore[assignment]
+            msg = f"expect scale_factor is float or Sequence[float], but get {type(scale_factor)}"
+            raise TypeError(msg)
```

Comment:
I find this very confusing even as code, how about adding Example to the docstring?